### ISSUE TITLE
[FEATURE]: Add HTTP proxy parameter to idrac_firmware to download update files via proxy.

### ISSUE THREAD

**Issue by Mixer9** [dell/dellemc-openmanage-ansible-modules#444]
**Describe the solution you'd like**
There needs to be an option for the idrac_firmware module to use a proxy to download update files. 

**Describe alternatives you've considered**
I tried setting the proxy environment variables for my user and for root, I thought maybe since it was using "become" that it might need the variables in roots environment, but alas that didn't work.  I also tried setting the proxy in the /etc/environment file but no joy, that didn't work either.

**Additional context**
I have tested that the proxy is needed in order to get to downloads.dell.com from my host.
Here is a snippet from the failure message.
```
"DisplayName": "OS COLLECTOR 2.1
A00"
"JobID": "JID_642106188711"
"JobStatus": "Critical"
"Message": "Unable to download Update Package."
"PackageName": "Diagnostics_Application_0G1JH_WN64_6.0_A00_01.EXE"
"PackagePath": "FOLDER06878175M/2/Diagnostics_Application_0G1JH_WN64_6.0_A00_01.EXE"
"PackageVersion": "6.0"
"RebootType": "HOST"
"Target": "DCIM:INSTALLED#802__OSCollector.Embedded.1"}
{"BaseLocation": "downloads.dell.com"
"ComponentID": null
"ComponentType": "FRMW"
"Criticality": "1"
"DisplayName": "PERC H730 Mini"
"JobID": "JID_642106084132"
"JobStatus": "Critical"
"Message": "Unable to download Update Package."
"PackageName": "SAS-RAID_Firmware_700GG_WN64_25.5.9.0001_A17.EXE"
"PackagePath": "FOLDER07217671M/1/SAS-RAID_Firmware_700GG_WN64_25.5.9.0001_A17.EXE"
"PackageVersion": "25.5.9.0001"
"RebootType": "HOST"
"Target": "DCIM:INSTALLED#301_C_RAID.Integrated.1-1"}
{"BaseLocation": "downloads.dell.com"
"ComponentID": null
"ComponentType": "FRMW"
"Criticality": "3"
```

**Community Note**

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) 
  to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, 
  they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

**Comment by anupamaloke** [comment]
@Mixer9, thanks for submitting this feature request. It has been added to backlog.

**Comment by anupamaloke** [comment]
Closed by #458
